don't use ssh for fetching  linphone submodule.

shareefalis opened the pull request:
- fixes issue with fetching from anonymous account.

shareefalis committed:
don't use ssh for fetching  linphone submodule.

* fixes issue with fetching from anonymous account.

shareefalis committed:
get rid of belledonne build status tracking.

* We are not using this status for tracking builds

shareefalis committed:
Add cmake as a required dependancy for bbuilding linphone

shareefalis committed:
update belle-sip submodule to dev-rtt version

ianblenke commented on README.md:
```diff
@@ -1,7 +1,5 @@
 # Linphone on iPhone
 
-[![Build Status](https://travis-ci.org/BelledonneCommunications/linphone-iphone.svg?branch=master)](https://travis-ci.org/BelledonneCommunications/linphone-iphone)
```
This begs the question of whether we really want to start forking our codebase that far from the linphone mainstream codebase. I see where you are going here, but this kind of change will likely get a merge reject if we PR back upstream. I'll talk about this with Peter.

ianblenke replied:
On second thought, we'll deal with this conflict when we go to merge this back upstream later with linphone-iphone proper.

ianblenke commented on README.md:
```diff
@@ -16,7 +14,7 @@ Linphone for iPhone depends on liblinphone SDK. This SDK is generated from makef
 * Using HomeBrew:
 
-        brew install autoconf automake pkg-config doxygen nasm gettext wget yasm optipng imagemagick coreutils intltool
+        brew install autoconf automake pkg-config doxygen nasm gettext wget yasm optipng imagemagick coreutils intltool cmake
```
Good catch. This is the kind of thing we'll want to contribute back upstream.

shareefalis committed:
Fix compile and link error when compiling the project.

Still need iOS libintl.dylib and libintl.a file for compiling.
Almost all the link error are gone.

shareefalis committed:
update linphone submodule

shareefalis committed:
Add instruction for installing xcode command line tools

shareefalis committed:
Remove librarys which were causing duplicate function linker errors.

* These librarys are plugin anyways

ianblenke commented on linphone.xcodeproj/project.pbxproj:
```diff
@@ -91,8 +89,6 @@
 		22A10F3B11F8960300373793 /* libortp.a in Frameworks */ = {isa = PBXBuildFile; fileRef = 220FAD2C10765B400068D98F /* libortp.a */; };
 		22A7D83E1B289EB30065E5D3 /* wizard_vtcsecure_existing.rc in Resources */ = {isa = PBXBuildFile; fileRef = 22A7D83D1B289EB30065E5D3 /* wizard_vtcsecure_existing.rc */; };
-		22AA8AFD13D7125600B30535 /* libx264.a in Frameworks */ = {isa = PBXBuildFile; fileRef = 22AA8AFB13D7125500B30535 /* libx264.a */; };
-		22AA8AFE13D7125600B30535 /* libmsx264.a in Frameworks */ = {isa = PBXBuildFile; fileRef = 22AA8AFC13D7125500B30535 /* libmsx264.a */; };
```
You seem to be removing x264 entirely as a library dependency.

shareefalis replied:
I don't know why but it was giving me a duplicate linker error. I think these libraries are meant to be dynamic/plug in based in nature.

ianblenke commented on linphone.xcodeproj/project.pbxproj:
```diff
@@ -1881,25 +1886,23 @@
 				225F6C081B380E0000A570DC /* libSKP_SILK_SDK.a in Frameworks */,
-				225F6C011B380DE600A570DC /* libxml2.a in Frameworks */,
 				C6F476531B27157D00F4C2A7 /* CoreLocation.framework in Frameworks */,
-				F0B026F31AA710AF00FF49F7 /* libiconv.dylib in Frameworks */,
```
It seems like libiconv is being removed, and that libintl is being added. Not that there is anything wrong with doing this, it's just something I didn't have to do with my builds here.

shareefalis replied:
Since I removed libiintl support from liblinphone. There was no harm in removing a library that is not used (linked) right now but my priority was to get it compiling/linking correctly so I can get the entire branch in good shape but anyways i agree with removing dependencies libraries is questionable at best. (I was getting pretty frustrated at the time when it kept failing at linker stage),

ianblenke commented on submodules/belle-sip:
```diff
@@ -1 +1 @@
-Subproject commit dd2d1ca8b3f56044f9ff85f0647184204ae382b2
```
This was a weird one alright. I had to essentially do the same thing myself. The submodule'd belle-sip version was older than the 0.3 that it needed to build.

ianblenke commented:
Looks good to me. +1 

Going to merge this now.
